FMM/meson.build

```diff
@@ -0,0 +1,19 @@
+project('FMM', 'c')
+
+m_dep = meson.get_compiler('c').find_library('m', required : false)
+
+executable(
+  'test_artUpdate',
+  ['test_artUpdate.c',
+   'SoSFunction.c', 'eik_grid.c', 'files_methods.c',
+   'linAlg.c', 'opti_method.c', 'priority_queue.c', 'coord.c',
+   'faces.c', 'facets.c', 'neighbors.c', 'triMesh_2D.c', 'fmm_2d.c', 'path.c'],
```

Comment:
You might want to compile most of these source files as a `libcommon = static_library('common', ....)` and then link both executables to it. That will avoid having to compile each file twice, once for each executable (possibly with different `c_args:`)

Reply:
Yup. 

Who are you? How come you're commenting on this PR?

Reply:
I sometimes comment on random Meson stuff I stumble across. :)

(I'm a core committer for Meson itself.)

Reply:
Ah, got it. Thanks for Meson! It's my current favorite piece of software. I can't wait for you to destroy CMake. :-)

@mtzmarianaa, you can see that I just made the change that @eli-schwartz suggested. The idea here is that since you have some code which is common to each of your executables, you can wrap the common stuff up into a "static library" which you can then link into each of your executables. With the rough draft that I had before, each of the files would need to be compiled separately for each executable. It doesn't really make a big difference here, because this project is quite small, but for larger projects you can save on build time and mental overhead by having things nicely organized into their constituent parts (modules). This is along the same lines as our conversation in my office the other day... It pays to be organized... otherwise randos on the internet will call you out. :-)